operators/src/processing/raster_vector_join/non_aggregated.rs

```diff
@@ -93,61 +93,68 @@ where
         aggregation_method: FeatureAggregationMethod,
         ignore_no_data: bool,
     ) -> Result<BoxStream<'a, Result<FeatureCollection<G>>>> {
-        call_on_generic_raster_processor!(raster_processor, raster_processor => {
-            Self::process_typed_collection_chunk(collection, raster_processor, new_column_name, query, ctx, aggregation_method, ignore_no_data).await
-        })
+        if !collection.is_empty() {
```

Comment:
The early return case should be the empty case and not the default case.

Reply:
yes,but if there is no intersection in time or space we also can just return the collection with a null column. If i handle the empty case as an early return, then i have to duplicate the code for that.

Reply:
do you think:

```

if bbox.is_none() or time.is_none() or collection.is_empty() {
// return early
}

bbox = bbox.unwrap()
time = time.unwrap()

// default case
```
is better? This would also calculate the bbox and the time intersection even if the collection is empty.
If you have an idea how refactor please tell me. We could crate a  new method  that we call but thats just more code for the same effect...

Reply:
I've added a suggestion below.
This should compile and successfully run the tests.

Reply:
while i think that the other way around it was less code i am fine with your suggestion.
It did not compile because of whitespaces that fmt was complaining about.